File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    for i in range(1, (brown-4)//2):
        if i*((brown-4)//2-i) == yellow:
            answer.append(i+2)
            answer.append((brown-4)//2-i+2) 
            break
    answer = sorted(answer, reverse=True)
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let answer = [];
    for (let i = 1; i < (brown - 4) / 2; i++) {
        if (i * ((brown - 4) / 2 - i) === yellow) {
            answer.push(i + 2);
            answer.push(((brown - 4) / 2 - i) + 2);
            break;
        }
    }
    answer.sort((a, b) => b - a);
    return answer;
}
